File: app/src/main/java/com/simon_eye/cardsscoring/LikhaGame.java
package com.simon_eye.cardsscoring;

import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.Toast;

/**
 * Created by rami_m on 5/26/2018.
 */


public class LikhaGame extends gameActivity {

    // define variables for LIKHA game
    protected static final int FINAL_SCORE = 101;  // First player to reach 101 or higher loses
    private static final int GAME_TITLE = R.string.likhaGameTitle;
    private static final int ROUND_TOTAL = 36;  // Round score total must be 36
    private static final int NUM_OF_PLAYERS = 4; // number of players in Likha
    private static final String PLAYER_INITIAL = "P"; // player Initial in Likha game

    public LikhaGame() {
        super(GAME_TITLE, NUM_OF_PLAYERS, PLAYER_INITIAL);

    }


    @Override
    public boolean gameEndCheck() {
        for (int i = 0; i < playerScores.length; i++) {
            if (playerScores[i] >= FINAL_SCORE) {
                displayGameResult(String.format("%s Lost", playerNames.get(i)));
                return true;
            }
        }
        return false;
    }

    @Override
    public boolean doubleTapAction(int autoFillField) {
        int remainingScore = 0;

        ViewGroup group = findViewById(R.id.currentScore);
        for (int i = 0, count = group.getChildCount(); i < count; ++i) {
            View view = group.getChildAt(i);
            if (view instanceof EditText) {
                String score = ((EditText) view).getText().toString();
                if (!score.equals("") && view.getId() != autoFillField) {
                    remainingScore += Integer.parseInt(score);
                }
            }
        }
        if (remainingScore > ROUND_TOTAL) {
            Toast.makeText(getApplicationContext(), "Invalid Input",
                    Toast.LENGTH_SHORT).show();
            return false;
        }
        remainingScore = ROUND_TOTAL - remainingScore;
        EditText scoreField = findViewById(autoFillField);
        scoreField.setText(String.valueOf(remainingScore));
        return true;
    }

    @Override
    public boolean scoreCheck(int[] tempScore) {
        // check if the values sum to 36
        int tempTotalScore = 0;

        // create int [] out of the string [] of scores
        for (int atempScore : tempScore) {
            tempTotalScore += atempScore;
        }
        // if the scores sum is not 36 return false
        if (tempTotalScore != ROUND_TOTAL) {
            Toast.makeText(getApplicationContext(), "Sum must be 36",
                    Toast.LENGTH_SHORT).show();
            return false;
        }
        submittedScore = tempScore;
        return true;
    }
}

File: app/src/main/java/com/simon_eye/cardsScoring/LikhaGame.kt
package com.simon_eye.cardsScoring

import android.view.ViewGroup
import android.widget.EditText
import android.widget.Toast
import kotlinx.android.synthetic.main.game_layout.*

/**
 * Created by rami_m on 5/26/2018.
 */


class LikhaGame : gameActivity(GAME_TITLE, NUM_OF_PLAYERS, PLAYER_INITIAL) {


    override fun gameEndCheck(): Boolean {
        for (i in 0 until playerScores.size) {
            if (playerScores[i] >= FINAL_SCORE) {
                displayGameResult(String.format("%s Lost", playerNames[i]))
                return true
            }
        }
        return false
    }

    override fun doubleTapAction(autoFillField: Int): Boolean {
        var remainingScore = 0

//        val currentScore = findViewById<ViewGroup>(R.id.currentScore)
        var i = 0
//        val count = currentScore.childCount
        while (i < currentScore.childCount) {
            val view = currentScore.getChildAt(i)
            if (view is EditText) {
                val score = view.text.toString()
                if (score != "" && view.getId() != autoFillField) {
                    remainingScore += Integer.parseInt(score)
                }
            }
            ++i
        }
        if (remainingScore > ROUND_TOTAL) {
            Toast.makeText(applicationContext, "Invalid Input",
                    Toast.LENGTH_SHORT).show()
            return false
        }
        remainingScore = ROUND_TOTAL - remainingScore
        val scoreField = findViewById<EditText>(autoFillField)
        scoreField.setText(remainingScore.toString())
        return true
    }

    override fun scoreCheck(tempScore: IntArray): Boolean {
        // check if the values sum to 36
        var tempTotalScore = 0

        // create int [] out of the string [] of scores
        for (atempScore in tempScore) {
            tempTotalScore += atempScore
        }
        // if the scores sum is not 36 return false
        if (tempTotalScore != ROUND_TOTAL) {
            Toast.makeText(applicationContext, "Sum must be 36",
                    Toast.LENGTH_SHORT).show()
            return false
        }
        submittedScore = tempScore
        return true
    }

    companion object {

        // define variables for LIKHA game
        protected val FINAL_SCORE = 101  // First player to reach 101 or higher loses
        private val GAME_TITLE = R.string.likhaGameTitle
        private val ROUND_TOTAL = 36  // Round score total must be 36
        private val NUM_OF_PLAYERS = 4 // number of players in Likha
        private val PLAYER_INITIAL = "P" // player Initial in Likha game
    }
}
